### PR TITLE
feat: WebDAV property presets (built-ins, tools, XML generation)

### PR DESCRIPTION
**Pull request by bmordue**
This PR introduces the Property Presets feature.

Key points:
- Built-in presets: basic, detailed, minimal.
- New tools: list_property_presets, get_property_preset.
- Extended dav_request with preset & additionalProperties for PROPFIND.
- XML generation helper with namespace prefixing & duplicate deduplication.
- Caching layer (TTL + mtime) merging user presets from DAV_PROPERTY_PRESETS_DIR.
- Tests covering built-ins, XML generation, merge logic.
- README updated (usage, examples, env vars).

### CODE REVIEW

File: src/presets/index.ts [hunk 1]
```diff
@@ -0,0 +1,200 @@
...
+  if (properties.length === 0 || properties.length > MAX_PROPERTIES_PER_PRESET) return null;
+  return {
+    name,
+    description: typeof preset.description === 'string' ? preset.description : undefined,
+    properties,
+    builtin: preset.builtin === true,
+  };
+}
+
+function loadUserPresets(): PropertyPreset[] {
```
[the remark this answers is not in the document]
Comment (bmordue) [reply src/presets/index.ts]: @copilot 

high

The loadUserPresets function uses synchronous file system operations (fs.existsSync, fs.readdirSync, fs.statSync, fs.readFileSync). These calls block the Node.js event loop, which can severely degrade server performance and responsiveness, especially if the presets directory contains many files. The cacheValid function has the same issue.

All file system operations should be converted to their asynchronous counterparts from the fs/promises module. This will require this function, and functions calling it (like getAllPresets), to become async.

File: src/presets/index.ts [hunk 2]
```diff
@@ -0,0 +1,200 @@
...
+      }
+    } catch (e) {
+      console.warn(`[presets] Failed to load ${file}: ${(e as Error).message}`);
+    }
+  }
+  if (all.length > MAX_PRESETS_TOTAL) {
+    console.warn(`[presets] Too many presets loaded (${all.length}), truncating to ${MAX_PRESETS_TOTAL}`);
+    return all.slice(0, MAX_PRESETS_TOTAL);
+  }
+  cache = { loadedAt: Date.now(), presets: [...BUILTIN_PRESETS, ...all], mtimes };
```
[the remark this answers is not in the document]
Comment (bmordue) [reply src/presets/index.ts]: @copilot Currently, user-defined presets do not override built-in presets with the same name. Because user presets are appended to the list after built-ins, getPreset (which uses find) will always return the built-in version. Users might expect to be able to customize or override built-in presets.

To fix this, you can use a Map to store presets, ensuring that user presets with conflicting names overwrite the built-in ones.

```js
  const presetMap = new Map<string, PropertyPreset>();
  BUILTIN_PRESETS.forEach(p => presetMap.set(p.name, p));
  all.forEach(p => presetMap.set(p.name, p)); // User presets overwrite built-ins
  cache = { loadedAt: Date.now(), presets: Array.from(presetMap.values()), mtimes };
  ```

File: src/tests/presets.test.ts [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+export function runAllTests() {
+  testBuiltinPresets();
+  testGenerateXml();
+  testMergeProperties();
+  console.log('All tests passed');
+}
+
+if (import.meta.url === `file://${process.argv[1]}`) {
+  runAllTests();
+}
```
[the remark this answers is not in the document]
Comment (bmordue) [reply src/tests/presets.test.ts]: @copilot The current test suite for presets is a good start, but it's quite minimal. It doesn't cover several critical parts of the new functionality, such as:

    Validation: The validatePreset function is not tested with invalid data (e.g., invalid names, missing properties, oversized presets).
    File Loading: The logic for loading user presets from JSON files, including handling of malformed files, is not tested.
    Caching: The cache invalidation logic (both TTL and mtime-based) is not covered.
    XML Generation Edge Cases: Tests for generatePropfindXml could be expanded to include multiple custom namespaces to ensure prefix generation works correctly.

Improving test coverage would increase confidence in the robustness and security of this new feature.
